src/executors.rs

```diff
@@ -132,3 +132,40 @@ impl AgnostikExecutor for BastionExecutor {
         bastion_executor::run::run(future, ProcStack::default())
     }
 }
+
+#[cfg(feature = "runtime_nostd")]
+pub(crate) struct NoStdExecutor;
+
+#[cfg(feature = "runtime_nostd")]
+impl NoStdExecutor {
+    pub fn new() -> Self {
+        NoStdExecutor {}
+    }
+}
+
+#[cfg(feature = "runtime_nostd")]
+impl AgnostikExecutor for NoStdExecutor {
+    fn spawn<F, T>(&self, _future: F) -> JoinHandle<T>
+    where
+        F: Future<Output = T> + Send + 'static,
+        T: Send + 'static,
+    {
+        panic!("no threads on no_std environments")
```

Comment:
Threads aren't necessary for spawning tasks, it's possible to have fully capable core-only executors, they're just going to necessarily be much more device specific than any using `std`.

Reply:
awesome, I thought someone might say something about this, I'm curious what people think should be done about these cases for `no_std` , a better message? no_std multithreading is a bit out of my knowledge base.

Reply:
:man_shrugging: I would say that `no_std` users should likely just supply their own, it's unlikely that there's a decent one-size-fits-all approach possible.

Reply:
Yah, I agree, i'm a web assembly dev and I don't even really have an option in this respect :)